### PR TITLE
Fix #3635 and #2694

### PR DESCRIPTION
**Pull request by andreynovikov**

### CODE REVIEW

File: ensure_dependencies.py [hunk 1]
```diff
@@ -304,6 +304,9 @@ def resolve_deps(repodir, level=0, self_update=True, overrideroots=None, skipdep
     parenttype = get_repo_type(repodir)
     _root = config.get("_root", {})
 
+    if parenttype is None:
```
Comment (snoack) [inline ensure_dependencies.py]: Instead adding two new lines, you could simply return `'hg'` instead `None` from `get_repo_types()`.

**Comment by snoack** [comment]
Thanks! Just one nit.

**Comment by andreynovikov** [comment]
What's the proper way to amend it?

**Comment by snoack** [comment]
How do you mean? I suggested to simply replace `return None` with `return "hg"` in `get_repo_type()`. I have no idea how I could be anymore specific. ;)

**Comment by andreynovikov** [comment]
No. I'm talking about pull request. How do I amend it? I'm not so familiar with git.

**Comment by palant** [comment]
@andreynovikov: You add more commits to the branch, these will automatically show up in the pull request.

**Comment by andreynovikov** [comment]
Updated

**Comment by snoack** [comment]
Thanks! LGTM and merged.

**Pull request closed without merging**